### PR TITLE
[tests][ci] Refactor ci

### PR DESCRIPTION
**Pull request by GOODBOY008**

### CODE REVIEW

**Comment by yuxiqian** [comment]
Thanks for @GOODBOY008's great work!

A little suggestion: is it possible to split jobs into sub-workflow files to organize them by categories? Current there are a little too much jobs, and most of them has only one sub-task `compile_and_test`.

截圖 2024-09-30 17 10 48

**Review by yuxiqian — COMMENTED** [review]
Seems Flink version specification isn't working as expected. Flink 1.20 test job is still running tests on Flink 1.17 (and all other versions).

截圖 2024-09-30 17 24 43

**Review by yuxiqian — APPROVED** [review]
Looks good.

Categorized folders aren't that important. Could be done later if it takes too much efforts to do now.

File: .github/workflows/flink_cdc_base.yml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+  flink-cdc-composer,\
+  flink-cdc-runtime"
+
+  MODULES_PIPELINE_CONNECTORS: "\
+  flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values,\
+  flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql,\
+  flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris,\
+  flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks,\
+  flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-kafka,\
+  flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon"
```
Comment (yuxiqian) [inline .github/workflows/flink_cdc_base.yml:52]: Somehow Elasticsearch connector tests have never been added into CI...

**Comment by Jiabao-Sun** [comment]
Thanks @GOODBOY008 for this great work.
How about using github action's matrix to control flink's version that can improve the coverage of the flink versions and simplify the Parameterized Tests.

**Pull request merged**